packages/liferay-js-insights/insights/dependencies.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Extracts import declarations from a given AST applying the following transforms:
+ * - Relative imports are discarded
+ * - Imports are grouped per origin into:
+ * 	- clay3
+ *  - react
+ *  - js
+ *  - metal
+ *  - clay2
+ *  - others
+ */
+function extractImports(ast) {
+	const EMPTY = [[], [], [], [], [], []];
+	const SOURCE_GROUPS = [
+		source => source.includes('@clayui'),
+		source =>
+			source.includes('react') ||
+			source.includes('prop-types') ||
+			source.includes('classnames'),
+		source => source.includes('frontend-js-web'),
+		source => source.includes('metal-') || source === 'metal',
+		source => source.includes('clay-') || source === 'clay',
+		_ => true // eslint-disable-line
+	];
+
+	function getGroup(source) {
+		return SOURCE_GROUPS.findIndex(group => group(source));
+	}
+
+	const externalImports = ast.program.body.filter(
+		node =>
+			node.type === 'ImportDeclaration' &&
+			!node.source.value.includes('./')
+	);
+
+	return [
+		externalImports.length,
+		...(externalImports.reduce((imports, node) => {
+			const group = getGroup(node.source.value);
+
+			imports[group] = [
+				...imports[group],
+				node.specifiers
+					.filter(specifier => specifier.local.name)
+					.map(specifier => specifier.local.name)
+			].reduce((memo, it) => memo.concat(it), []); // Not using flatMap to support Node.js 10;;;
+
+			return imports;
+		}, EMPTY) || EMPTY)
+	];
+}
+
+/**
+ *
+ */
+module.exports = async function(ast, insights) {
+	const [count, clay3, react, js, metal, clay2, others] = extractImports(ast);
```

Comment:
This is also order-sensitive, which seems a bit brittle, especially because it is far away from the `SOURCE_GROUPS` declaration. Did you think about using an object or `Map` instead?

Reply:
I already forgot why I did it this way... probably had a bad reason for it 😂 

I'll probably take a look on the plane back :D